src/Codata/Colist/Properties.agda

```diff
@@ -8,25 +8,326 @@
+drop-fromList-++-identity []       bs = Eq.refl
+drop-fromList-++-identity (a ∷ as) bs = drop-fromList-++-identity as bs
+
+drop-fromList-++-≤ : ∀ (as : List A) bs {m} → m ℕ.≤ List.length as →
+                     drop m (fromList as ++ bs) ≡ fromList (List.drop m as) ++ bs
+drop-fromList-++-≤ []       bs z≤n     = Eq.refl
+drop-fromList-++-≤ (a ∷ as) bs z≤n     = Eq.refl
+drop-fromList-++-≤ (a ∷ as) bs (s≤s p) = drop-fromList-++-≤ as bs p
+
+drop-fromList-++-≥ : ∀ (as : List A) bs {m} → m ℕ.≥ List.length as →
+                     drop m (fromList as ++ bs) ≡ drop (m ℕ.∸ List.length as) bs
+drop-fromList-++-≥ []       bs z≤n     = Eq.refl
+drop-fromList-++-≥ (a ∷ as) bs (s≤s p) = drop-fromList-++-≥ as bs p
+
+drop-⁺++-identity : ∀ (as : List⁺ A) bs →
+                    drop (List⁺.length as) (as ⁺++ bs) ≡ bs .force
+drop-⁺++-identity (a ∷ as) bs = drop-fromList-++-identity as (bs .force)
+
+------------------------------------------------------------------------
+-- Properties of cotake
+
+length-cotake : ∀ n (as : Stream A ∞) → i coℕᵇ.⊢ length (cotake n as) ≈ n
+length-cotake zero    as       = zero
+length-cotake (suc n) (a ∷ as) =
+  suc λ where .force → length-cotake (n .force) (as .force)
+
+map-cotake : ∀ (f : A → B) n as →
+             i ⊢ map f (cotake n as) ≈ cotake n (Stream.map f as)
+map-cotake f zero    as       = []
+map-cotake f (suc n) (a ∷ as) =
+  Eq.refl ∷ λ where .force → map-cotake f (n .force) (as .force)
+
+------------------------------------------------------------------------
+-- Properties of chunksOf
+
+module Map-ChunksOf (f : A → B) n where
+
+  open ChunksOf n using (chunksOfAcc)
+
+  map-chunksOf : ∀ as →
+    i coWriterᵇ.⊢ Cowriter.map (Vec.map f) (Vec≤.map f) (chunksOf n as)
+                ≈ chunksOf n (map f as)
+  map-chunksOfAcc : ∀ m as {k≤ k≡ k≤′ k≡′} →
+    (∀ vs → Vec≤.map f (k≤ vs) ≡ k≤′ (Vec≤.map f vs)) →
+    (∀ vs → Vec.map f (k≡ vs) ≡ k≡′ (Vec.map f vs)) →
+    i coWriterᵇ.⊢ Cowriter.map (Vec.map f) (Vec≤.map f)
+                    (chunksOfAcc m k≤ k≡ as)
+                ≈ chunksOfAcc m k≤′ k≡′ (map f as)
+
+  map-chunksOf as = map-chunksOfAcc n as (λ vs → Eq.refl) (λ vs → Eq.refl)
+
+  map-chunksOfAcc zero    as       eq-≤ eq-≡ =
+      eq-≡ [] ∷ λ where .force → map-chunksOf as
+  map-chunksOfAcc (suc m) []       eq-≤ eq-≡ = coWriterᵇ.[ eq-≤ Vec≤.[] ]
+  map-chunksOfAcc (suc m) (a ∷ as) eq-≤ eq-≡ =
+    map-chunksOfAcc m (as .force) (eq-≤ ∘ (a Vec≤.∷_)) (eq-≡ ∘ (a Vec.∷_))
+
+open Map-ChunksOf using (map-chunksOf) public
+
+------------------------------------------------------------------------
+-- Properties of fromList
+
+fromList-++ : (as bs : List A) →
+              i ⊢ fromList (as List.++ bs) ≈ fromList as ++ fromList bs
+fromList-++ []       bs = refl
+fromList-++ (a ∷ as) bs = Eq.refl ∷ λ where .force → fromList-++ as bs
+
+fromList-scanl : ∀ (c : B → A → B) n as →
+                 i ⊢ scanl c n (fromList as) ≈ fromList (List.scanl c n as)
```

Comment:
If we're calling it `fromList-scanl` can we reverse the equality to match the name? (hence making it consistent with the other `fromList` lemmas?)